src/flask_cognito_lib/services/cognito_svc.py

```diff
@@ -96,6 +97,116 @@ def exchange_code_for_token(
             "code_verifier": code_verifier,
         }
 
+        return self._request_token(data)
+
+    def refresh_token(
+        self,
+        refresh_token: str,
+    ) -> CognitoTokenResponse:
+        """
+        Exchange a refresh token for a new set of tokens
+
+        Parameters:
+        -----------
+        refresh_token : str
+            The refresh token to exchange for a new set of tokens
+
+        Returns:
+        --------
+        CognitoTokenResponse
+            A dataclass that holds the token response from Cognito
+
+        Raises:
+        -------
+        CognitoError
+            If the request to the endpoint fails
+            If the endpoint returns an error code
+        """
+        data = {
+            "grant_type": "refresh_token",
+            "client_id": self.cfg.user_pool_client_id,
+            "refresh_token": refresh_token,
+        }
+
+        return self._request_token(data)
+
+    def revoke_refresh_token(
+        self,
+        refresh_token: str,
+    ) -> None:
+        """
+        Revoke a refresh token
+
+        Parameters:
+        -----------
+        refresh_token : str
+            The refresh token to revoke
+
+        Raises:
+        -------
+        @see https://docs.aws.amazon.com/cognito/latest/developerguide/revocation-endpoint.html#revoke-sample-response
+
+        CognitoError
+            If the token isn't present in the request or if the feature is disabled for the app client
+            If the token that Amazon Cognito sent in the revocation request isn't a refresh token
+            If the client credentials aren't valid
+        """  # noqa
+        data = {
+            "token": refresh_token,
+        }
+
+        self._request(url=self.cfg.revoke_endpoint, data=data)
+
+    def _request_token(self, data: dict) -> CognitoTokenResponse:
```

Comment:
type hint for dict should specify what it contains, I think it will always be `dict[str, str]` though technically I think requests would accept `dict[str, Any]` and stringify it

Reply:
I went with `dict[str, str]` as we have a specific case so can limit possible payload.